routes/signup.tsx

```diff
@@ -18,6 +20,7 @@ export default function SignupPage(props: PageProps) {
           {errorMessage === "User already registered" && (
             <Notice>{errorMessage}</Notice>
           )}
+          <OAuthLoginButton provider="github" icon={Icons.GitHub} />
```

Comment:
I think the button label needs to be more flexible to support other providers.

How about the interface like the below?

```js
<OAuthLoginButton provider="github">
  <Icons.GitHub class="inline mr-2" />
  Log in with GitHub
</OAuthLoginButton>
```

Reply:
`OAuthLoginButton` does support other providers. Is there something about it that appears that it might not?

Reply:
On 2nd thought, you're right. `OAuthLoginButton` did too much. I've refactored it.